pkg/supportbundle/spec.go

```diff
@@ -144,7 +144,7 @@ func CreateRenderedSpec(app apptypes.AppType, sequence int64, kotsKinds *kotsuti
 	renderedSpec = b.Bytes()
 
 	secretName := GetSpecSecretName(app.GetSlug())
-	existingSecret, err := clientset.CoreV1().Secrets(util.PodNamespace).Get(context.TODO(), secretName, metav1.GetOptions{})
+	existingSecret, err := GetSpecSecretsMatchingLabel(clientset, kotstypes.LabelsToStrings(kotstypes.GetTroubleshootLabels()), util.PodNamespace, SpecDataKey)
```

Comment:
How might this work in an environment that was running an older version of kots and then upgraded to this version.  The older version may have an existing secret, but may not have the newer troubleshoot label (`troubleshoot.io/kind=support-bundle`).  Would this function return `kuberneteserrors.IsNotFound(err) = true` and kots would try to create a new support bundle secret with the same name (and may fail because it already exists, but with different labels)?

Reply:
Could it work if we also use `GetKotsadmLabels()` in addition to the troubleshoot labels?  Or would that select only secrets that have _both_ instead of _either_?

Reply:
Here is output from a test of this.  The support bundle failed to run:

Existing secret in cluster:
```
apiVersion: v1
data:
  support-bundle-spec: ...
kind: Secret
metadata:
  creationTimestamp: "2022-07-18T16:02:48Z"
  labels:
    kots.io/backup: velero
    kots.io/kotsadm: "true"
  name: kotsadm-craig-helm-supportbundle
  namespace: cbodonnell
  resourceVersion: "346040016"
  uid: 16f3563f-c931-47e3-850b-95663a5aee0d
type: Opaque
```

Error:
```
{"level":"error","ts":"2023-01-19T15:26:09Z","msg":"could not generate support bundle dependencies: failed to create rendered support bundle spec: failed to create support bundle secret: secrets \"kotsadm-craig-helm-supportbundle\" already exists"}
```

Reply:
Great catch! Yes, I missed that part. Now I have added AddLabelExistingSpecSecret function to add the troubleshoot labels if the old secret spec exists. It will make sure that migrate all the existing spec secrets will new troubleshoot lablel. Tested in Okteto, looks ok. Could you help to double confirm? Thx!